### PR TITLE
READY: Replace twisted with asyncio

### PR DESCRIPTION
**Pull request by egbertbouman**

### CODE REVIEW

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,4 @@
 	url = https://github.com/Tribler/py-ipv8.git
 [submodule "Tribler/anydex"]
 	path = Tribler/anydex
-	url = https://github.com/Tribler/anydex-core.git
```
Comment (qstokkink) [inline .gitmodules]: This should be the main Tribler repo

File: Tribler/Core/Modules/restapi/channels_endpoint.py [hunk 2]
```diff
@@ -164,105 +94,65 @@ def render_GET(self, request):
...
+
+            if not target_collection and not personal_root:
+                return RESTResponse({"error": "Target channel not found"}, status=HTTP_NOT_FOUND)
+            results_list = []
+            for entry in request_parsed:
+                public_key, id_ = database_blob(unhexlify(entry["public_key"])), entry["id"]
+                source = self.session.lm.mds.ChannelNode.get(public_key=public_key, id_=id_)
+                if not source:
+                    return RESTResponse({"error": "Source entry not found"}, status=HTTP_BAD_REQUEST)
+                # We must upgrage Collections to Channels when moving them to root channel, and, vice-versa,
```
Comment (qstokkink) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: I know this is not your change, but could you please make this say "upgrade"?

Comment (egbertbouman) [reply Tribler/Core/Modules/restapi/channels_endpoint.py]: Will do.

File: .gitmodules (hunk 1 above)
Comment (egbertbouman) [reply .gitmodules]: I agree, but unfortunately https://github.com/Tribler/anydex-core/pull/30 still needs to be merged.

Comment (devos50) [reply .gitmodules]: I have created a separate `asyncio` [branch](https://github.com/Tribler/anydex-core/tree/asyncio) on the upstream repo, so this one can point to that branch, without changing the submodule URL 👍

Comment (egbertbouman) [reply .gitmodules]: The PR was already pointing to a pre-existing asyncio branch as far as I know, but thanks for merging!

**Comment by ichorid** [comment]
retest this please

**Pull request merged**